hs_tools_resource/app_launch_helper.py

```diff
@@ -92,24 +91,25 @@ def _get_app_tool_info(request_obj, resource_obj, tool_res_obj, open_with=False)
     if tool_res_obj.metadata.supported_file_extensions:
         file_extensions = tool_res_obj.metadata.supported_file_extensions.value
 
-    if (tool_url_resource_new is not None) or \
-            (tool_url_agg_new is not None) or \
-            (tool_url_file_new is not None):
-        tl = {'title': str(tool_res_obj.metadata.title.value),
-              'res_id': tool_res_obj.short_id,
-              'icon_url': tool_icon_url,
-              'url': tool_url_resource_new,
-              'url_aggregation': tool_url_agg_new,
-              'url_file': tool_url_file_new,
-              'openwithlist': is_open_with_app,
-              'approved': is_approved_app,
-              'agg_types': agg_types,
-              'file_extensions': file_extensions
-              }
-
-        return is_open_with_app, tl
+    if is_open_with_app or is_approved_app:
+        if (tool_url_resource_new is not None) or \
+                (tool_url_agg_new is not None) or \
+                (tool_url_file_new is not None):
+            tl = {'title': str(tool_res_obj.metadata.title.value),
+                  'res_id': tool_res_obj.short_id,
+                  'icon_url': tool_icon_url,
+                  'url': tool_url_resource_new,
+                  'url_aggregation': tool_url_agg_new,
+                  'url_file': tool_url_file_new,
+                  'openwithlist': is_open_with_app,
+                  'approved': is_approved_app,
+                  'agg_types': agg_types,
+                  'file_extensions': file_extensions
+                  }
+
+            return tl
```

Comment:
@sblack-usu adding ```is_open)with_app or is_approved_app``` may be problematic since an empty list would be returned if this added condition is False. Is this intended? Would ```openwithlist``` value in returned dictionary always true with this newly added condition?

Reply:
@sblack-usu I got it after reviewing the test changes, i.e., it is supposed to return None if the added condition is False. I still suspect```openwithlist``` key value would always be True with this change, though. If so, perhaps this key can be removed?

Reply:
You raise a good point and I think you're right.  I think the same could be said about the approved key, there's no reason to pass it up to the frontend.

Reply:
removed both keys